src/server/dfly_main.cc

```diff
@@ -323,13 +339,29 @@ bool RunEngine(ProactorPool* pool, AcceptServer* acceptor) {
   auto port = GetFlag(FLAGS_port);
   auto mc_port = GetFlag(FLAGS_memcache_port);
   string unix_sock = GetFlag(FLAGS_unixsocket);
+  mode_t unix_socket_perm;
+  string perm_str = GetFlag(FLAGS_unixsocketperm);
+  if (perm_str.empty()) {
+    // get umask of running process.
+    mode_t umask_val = umask(0);
+    umask(umask_val);
+    unix_socket_perm = 0777 & ~umask_val;
```

Comment:
It took me a a while to understand why the inverse. 
I did not know about `umask`, maybe worth adding a comment explaining that `umask_val` says which bits will be turned off